### PR TITLE
Fix undefined variables for color maps

### PR DESCRIPTION
**Pull request by kevinchern**
For this issue https://github.com/dwavesystems/dwave-networkx/issues/232

### CODE REVIEW

**Review by boothby — APPROVED** [review]

File: dwave_networkx/drawing/qubit_layout.py [hunk 1]
```diff
@@ -143,17 +125,16 @@ def node_color(v):
...
-            vmin = -1 * vmag
-
-        if vmax is None:
-            vmax = vmag
 
-        if edge_vmin is None:
-            edge_vmin = -1 * vmag
+        # since we're applying the colormap here, matplotlib throws warnings if
+        # we provide these arguments and it doesn't use them.
+        cmap = kwargs.pop('cmap', plt.get_cmap('coolwarm'))
```
Comment (randomir) [inline dwave_networkx/drawing/qubit_layout.py:131]: Is a change in semantics intended here? Consider the case when user supplies `cmap=None`.

File: dwave_networkx/drawing/qubit_layout.py [hunk 2]
```diff
@@ -143,17 +125,16 @@ def node_color(v):
...
+        vmin = kwargs.pop('vmin', -1 * vmag)
+        vmax = kwargs.pop('vmax', vmag)
 
-        if edge_vmax is None:
-            edge_vmax = vmag
+        edge_cmap = kwargs.pop('edge_cmap', plt.get_cmap('coolwarm'))
+        edge_vmin = kwargs.pop('edge_vmin', -1 * vmag)
+        edge_vmax = kwargs.pop('edge_vmax', vmag)
```
Comment (randomir) [inline dwave_networkx/drawing/qubit_layout.py:137]: Actually, the same question applies to all of these.

**Review by randomir — COMMENTED** [review]
Since you're already touching `linear_biases`/`quadratic_biases` here, would you mind fixing their [mutable default argument](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)?

Also, adding (at least some smoke) tests for `draw_qubit_graph` that verify the fix would be nice.

File: dwave_networkx/drawing/qubit_layout.py (hunk 1 above)
Comment (kevinchern) [reply dwave_networkx/drawing/qubit_layout.py:131]: The semantics should be the same. In the previous commit, we have
```
if linear_biases:
    cmap = kwargs.pop('cmap', None)
...
if cmap is None:
    cmap = plt.get_cmap('coolwarm')
```
which would have the same behaviour. The issue is if `linear_biases` was never supplied, then the second conditional throws an error as `cmap` is undefined.

**Comment by kevinchern** [comment]
I'll put in some tests

File: dwave_networkx/drawing/qubit_layout.py (hunk 1 above)
Comment (randomir) [reply dwave_networkx/drawing/qubit_layout.py:134]: No, previously if the function was called with `cmap=None` (explicitly set), you would end up with `cmap = plt.get_cmap('coolwarm')` (assuming it doesn't fail).

Now, if user explicitly sets `cmap=None`, the final value will be `cmap = None`.

Comment (kevinchern) [reply dwave_networkx/drawing/qubit_layout.py:134]: AH, OK I misunderstood.

File: dwave_networkx/drawing/qubit_layout.py (hunk 2 above)
Comment (kevinchern) [reply dwave_networkx/drawing/qubit_layout.py]: Fixed with https://github.com/dwavesystems/dwave-networkx/pull/233/files#diff-344ce7a800a7c461ff087cf0103c3ba81b6af1935497c0b179431a811faee391R134-R140

File: tests/test_qubit_layout.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    import numpy as np
+except ImportError:
+    np = False
+
+_display = os.environ.get('DISPLAY', '') != ''
+
+
+@unittest.skipUnless(np and plt, "No numpy or matplotlib")
+class TestDrawing(unittest.TestCase):
+    @unittest.skipUnless(_display, " No display found")
```
Comment (randomir) [inline tests/test_qubit_layout.py:35]: Not sure you need this if using non-interactive matplotlib backend. Btw, which backend is used here? Hopefully not interactive..

File: dwave_networkx/drawing/qubit_layout.py [hunk 4]
```diff
@@ -143,47 +128,37 @@ def node_color(v):
...
-
-        if vmax is None:
-            vmax = vmag
 
-        if edge_vmin is None:
-            edge_vmin = -1 * vmag
+        # since we're applying the colormap here, matplotlib throws warnings if
+        # we provide these arguments and it doesn't use them.
+        cmap = kwargs.pop('cmap', None) or plt.get_cmap('coolwarm')
+        vmin = kwargs.pop('vmin', None) or -1 * vmag
```
Comment (randomir) [inline dwave_networkx/drawing/qubit_layout.py:135]: `or` might not be what you want here. E.g. if user explicitly sets `vmin=0`, you override it to `-vmag`.

Comment (randomir) [reply dwave_networkx/drawing/qubit_layout.py:135]: When I say "here", I mean in this line, and a few other lines below.

File: tests/test_qubit_layout.py (hunk 3 above)
Comment (kevinchern) [reply tests/test_qubit_layout.py:35]: I added this for consistency with existing tests, e.g., https://github.com/dwavesystems/dwave-networkx/blob/main/tests/test_pegasus_layout.py#L68-L104

**Pull request merged**